### ISSUE TITLE
Error loading alignment track for unknown contigs

### ISSUE THREAD

**Issue by amwenger** [igvteam/igv#412]
The snapshot and 3.0beta versions have two similar issues when loading BAM alignment tracks:
* For a small genome with multiple contigs (e.g. _C. elegans_) that is zoomed to "All chromosome" level, loading a BAM produces the popup "Error encountered query alignments: java.lang.IllegalArgumentException: Invalid reference index -1".  [sample dataset for C. elegans WS201](https://gist.github.com/amwenger/64e6aae65dfe3ccb489befd623f09901)
* For other genomes, if a chromosome is missing from the header, then zooming to a region on that chromosome produces the same error. [sample dataset for hg19 with chrX missing from header; zoom to chrX](https://gist.github.com/amwenger/363e79a1eca827b758b4b2dcfebc18da)

The stack trace is:
```
ERROR [2017-05-15 15:41:32,389]  [AlignmentTileLoader.java:282] [pool-2-thread-2]  Error loading alignment data
java.lang.IllegalArgumentException: Invalid reference index -1
    at htsjdk.samtools.QueryInterval.<init>(QueryInterval.java:24)
    at htsjdk.samtools.SamReader$PrimitiveSamReaderToSamReaderAdapter.query(SamReader.java:504)
    at org.broad.igv.sam.reader.BAMReader.query(BAMReader.java:166)
    at org.broad.igv.sam.AlignmentTileLoader.loadTile(AlignmentTileLoader.java:151)
    at org.broad.igv.sam.AlignmentDataManager.loadInterval(AlignmentDataManager.java:309)
    at org.broad.igv.sam.AlignmentDataManager.load(AlignmentDataManager.java:284)
    at org.broad.igv.sam.AlignmentTrack.load(AlignmentTrack.java:397)
    at org.broad.igv.lists.Preloader.lambda$load$2(Preloader.java:103)
    at java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1626)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
    at java.lang.Thread.run(Thread.java:745)
INFO [2017-05-15 15:41:32,391]  [MessageUtils.java:74] [pool-2-thread-2]  <html>Error encountered querying alignments: java.lang.IllegalArgumentException: Invalid reference index -1
```

**Comment by jrobinso** [comment]
Thanks for the sample dataset,  have you tried this on 2.3.X?

Jim

**Comment by amwenger** [comment]
The 2.3.X version works fine for these tests.  Thanks for taking a look; I tried to hunt it down without luck.

**Comment by jrobinso** [comment]
Could be a change in the htsjdk,  but I will find it.  Thanks for the
report.

**Comment by jrobinso** [comment]
Fixed with commite f12513c1586397c84b6c6ce5896e98162f538183
The htsjdk change behavior for this case, previously it returned an empty iterator, now it throws an exception.  Exception is now caught and result of query is an empty iterator.

**Comment by amwenger** [comment]
Thank you, Jim.  Confirmed that the fix works for me.
